internal/worker/externalcontrollerupdater/externalcontrollerupdater_test.go

```diff
@@ -181,6 +182,46 @@ func (s *ExternalControllerUpdaterSuite) TestWatchExternalControllersErrorRestar
 	workertest.CleanKill(c, w)
 }
 
+func (s *ExternalControllerUpdaterSuite) TestWatchExternalControllersNotSupported(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	ch := make(chan []string, 1)
+	extCtrlWatcher := watchertest.NewMockStringsWatcher(ch)
+	ch <- []string{coretesting.ControllerTag.Id()}
+
+	s.client.EXPECT().WatchExternalControllers().Return(extCtrlWatcher, nil)
+	info := &crossmodel.ControllerInfo{
+		ControllerTag: coretesting.ControllerTag,
+		Alias:         "alias",
+		Addrs:         []string{"10.6.6.6"},
+		CACert:        coretesting.CACert,
+	}
+	s.client.EXPECT().ExternalControllerInfo(coretesting.ControllerTag.Id()).Return(info, nil)
+
+	notSupportedErr := &params.Error{Code: params.CodeAccessRequired}
+	started := make(chan struct{})
+
+	w, err := externalcontrollerupdater.New(s.client, func(*api.Info) (externalcontrollerupdater.ExternalControllerWatcherClientCloser, error) {
+		defer close(started)
+		return nil, notSupportedErr
+	}, s.clock)
+	c.Assert(err, jc.ErrorIsNil)
+	defer workertest.CleanKill(c, w)
```

Comment:
`CheckKilled`

Reply:
I'm a bit confused about this one. Pretty sure CleanKill is what we want here to cleanup the worker at the end of the test.

Reply:
Yeah exactly you've described the problem, but from the test we don't have access to the worker that has stopped.